src/test/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingTest.java

```diff
@@ -101,5 +103,76 @@ public void TranslateTriplePatternTest() {
 		
 		return new Pair<>(testSet, expectedRes);
 	}
+	*/
+	
+	@Test
+	public void TranslateSolutionMappingTest() {
+		final Set<Triple> testSet = new HashSet<>();
+		Node s = NodeFactory.createURI("a");
+		Node p = OWL.equivalentClass.asNode();
+		Node o = NodeFactory.createURI("n");
+		testSet.add(new Triple(s, p ,o));
+		
+		s = NodeFactory.createURI("b");
+		p = OWL.unionOf.asNode();
+		o = NodeFactory.createBlankNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = o;
+		p = RDF.first.asNode();
+		o = NodeFactory.createURI("c");
+		testSet.add(new Triple(s, p ,o));
+		
+		p = RDF.rest.asNode();
+		o = NodeFactory.createBlankNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = o;
+		p = RDF.first.asNode();
+		o = NodeFactory.createURI("n");
+		testSet.add(new Triple(s, p ,o));
+		
+		p = RDF.rest.asNode();
+		o = RDF.nil.asNode();
+		testSet.add(new Triple(s, p ,o));
+		
+		s = NodeFactory.createURI("c");
+		p = OWL.equivalentClass.asNode();
+		o = NodeFactory.createURI("m");
+		testSet.add(new Triple(s, p ,o));
+		
+		final VocabularyMapping vm = new VocabularyMappingImpl(testSet);
+		
+		final BindingBuilder testBuilder = BindingBuilder.create();
+		Var testVar = Var.alloc("v");
+		Node testNode = NodeFactory.createURI("n");
+		testBuilder.add(testVar, testNode);
```

Comment:
Instead of creating this variables and reassigning them, I would prefer if you create separate vars for each `add` or simply avoid them and just do:
```suggestion
		testBuilder.add( Var.alloc("v"), NodeFactory.createURI("n"));
```